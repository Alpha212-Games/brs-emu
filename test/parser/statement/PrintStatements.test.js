const Parser = require("../../../lib/parser");
const { Lexeme } = require("../../../lib/Lexeme");
const BrsError = require("../../../lib/Error");

const { token, EOF } = require("../ParserTests");

describe("parser", () => {
    afterEach(() => BrsError.reset());

    describe("print statements", () => {
        it("parses print statements", () => {
            let parsed = Parser.parse([
                token(Lexeme.Print),
                token(Lexeme.String, "Hello, world"),
                EOF
            ]);

            expect(parsed).toBeDefined();
            expect(parsed).not.toBeNull();
            expect(parsed).toMatchSnapshot();
        });
    });
});