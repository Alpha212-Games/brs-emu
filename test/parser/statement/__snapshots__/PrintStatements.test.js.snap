// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser print statements parses print statements 1`] = `
Array [
  Print {
    "expression": Literal {
      "value": "Hello, world",
    },
  },
]
`;
